File: solutions/ruby/armstrong-numbers/1/armstrong_numbers.rb
module ArmstrongNumbers
  def self.include?(number)
    return true if number < 10 
    sum = 0 
    digits = number.to_s.chars.map(&:to_i)
    digits_count = digits.size 

    digits.each {|digit| sum += digit ** digits_count }

    sum == number 
  end
end

File: solutions/cpp/armstrong-numbers/1/armstrong_numbers.cpp
#include "armstrong_numbers.h"
#include <cmath>
#include <vector>
using namespace std; 

namespace armstrong_numbers {
    bool is_armstrong_number(int number) {
        vector<int> digits;
        number = abs(number);
        auto copy{number};
        if (copy == 0) 
            digits.push_back(0);
    
        while (copy > 0) {
            digits.push_back(copy % 10);
            copy /= 10; 
        }
    
        auto n = digits.size();
        int sum{0};
        for (const auto d : digits) {
            sum += pow(d, n);
        }
    
        return sum == number;        
    }
}  // namespace armstrong_numbers
